src/cpp/StatisticsBackend.cpp

```diff
@@ -16,16 +16,137 @@
  * @file StatisticsBackend.cpp
  */
 
+#include <fstream>
+
+#include <fastdds/dds/domain/DomainParticipantFactory.hpp>
+#include <fastdds/dds/domain/DomainParticipant.hpp>
+#include <fastdds/dds/domain/qos/DomainParticipantQos.hpp>
+#include <fastdds/dds/topic/Topic.hpp>
+#include <fastdds/dds/topic/TopicDescription.hpp>
+#include <fastdds/dds/topic/qos/TopicQos.hpp>
+#include <fastdds/dds/subscriber/Subscriber.hpp>
+#include <fastdds/dds/subscriber/DataReader.hpp>
+#include <fastdds/dds/subscriber/qos/SubscriberQos.hpp>
+#include <fastdds/dds/subscriber/qos/DataReaderQos.hpp>
+#include <fastdds/dds/core/status/StatusMask.hpp>
+#include <fastdds/statistics/topic_names.hpp>
+
 #include <fastdds-statistics-backend/StatisticsBackend.hpp>
 #include <fastdds-statistics-backend/types/JSONTags.h>
-
 #include <database/database.hpp>
+#include <database/database_queue.hpp>
+#include <subscriber/StatisticsParticipantListener.hpp>
+#include <subscriber/StatisticsReaderListener.hpp>
+#include <topic_types/typesPubSubTypes.h>
+
+#include "Monitor.hpp"
 #include "StatisticsBackendData.hpp"
 
+using namespace eprosima::fastdds::dds;
+using namespace eprosima::fastdds::statistics;
+
 namespace eprosima {
 namespace statistics_backend {
 
 
+static const char* topics[] =
+{
+    HISTORY_LATENCY_TOPIC,
+    NETWORK_LATENCY_TOPIC,
+    PUBLICATION_THROUGHPUT_TOPIC,
+    SUBSCRIPTION_THROUGHPUT_TOPIC,
+    RTPS_SENT_TOPIC,
+    RTPS_LOST_TOPIC,
+    RESENT_DATAS_TOPIC,
+    HEARTBEAT_COUNT_TOPIC,
+    ACKNACK_COUNT_TOPIC,
+    NACKFRAG_COUNT_TOPIC,
+    GAP_COUNT_TOPIC,
+    DATA_COUNT_TOPIC,
+    PDP_PACKETS_TOPIC,
+    EDP_PACKETS_TOPIC,
+    DISCOVERY_TOPIC,
+    SAMPLE_DATAS_TOPIC,
+    PHYSICAL_DATA_TOPIC
+};
+
+void find_or_create_topic_and_type(
+        std::shared_ptr<details::Monitor> monitor,
+        const std::string& topic_name,
+        const TypeSupport& type)
+{
+    // Find if the topic has been already created and if the associated type is correct
+    TopicDescription* topic_desc = monitor->participant->lookup_topicdescription(topic_name);
+    if (nullptr != topic_desc)
+    {
+        if (topic_desc->get_type_name() != type->getName())
+        {
+            throw Error(topic_name + " is not using expected type " + type->getName() +
+                          " and is using instead type " + topic_desc->get_type_name());
+        }
+        monitor->topics[topic_name] = dynamic_cast<Topic*>(topic_desc);
```

Comment:
It is a good idea in any case to catch the exception and rethrow with a custom exception. However, I guess this will not be covered in current tests. 
I added a TODO nonetheless, because I'm not completely sure we should or should not support a TopicDescription that has the same name but is not a simple Topic. At this moment, at least, we do not support it because we cannot store it into the `topics` map